src/main.c

```diff
@@ -451,12 +457,20 @@ typedef struct {
 //TODO: Clean this up to use unions...
 sb_emu_state_t emu_state = { .joy.solar_sensor=0.5};
 #define SE_MAX_CONST(A,B) ((A)>(B)? (A) : (B) )
+#define SE_RC_BUFFER_SIZE 32768
 typedef union{
-  sb_gb_t gb;
-  gba_t gba;  
-  nds_t nds;
+  struct {
+    union {
+      sb_gb_t gb;
+      gba_t gba;  
+      nds_t nds;
+    };
+#ifdef ENABLE_RETRO_ACHIEVEMENTS
+    uint8_t rc_buffer[SE_RC_BUFFER_SIZE]; // buffer for RetroAchievements state, should be more than enough
```

Comment:
How do we know how large this needs to be?

Reply:
You're right we don't know at compile time, it needs to allocate a buffer of rc_client_progress_size bytes

Reply:
Doing so is a bit annoying cus it integrates less smoothly with the rewind/state api, and 32 KiB hasn't failed me yet on the amount of state rcheevos needs, and there's a check in place, see `retro_achievements_capture_state`

Reply:
Okay, thats fine. But lets bump it to 128KB just to be safe. Also, I just noticed a small bug. The rewind and save state functionality only record the first `se_get_core_size()` bytes of `se_core_state_t`. So, for this to be correct, rc_buffer needs to be moved to the front of the struct, and `se_get_core_size()` needs to be updated to incorporate the extra bytes for the rc_buffer.

Reply:
Oh, I see you are manually managing that section elsewhere in the code. So, I guess that should be okay.